lib/widgets/subscription_list.dart

```diff
@@ -0,0 +1,239 @@
+}
+
+class _SubscriptionListPageState extends State<SubscriptionListPage> with PerAccountStoreAwareStateMixin<SubscriptionListPage> {
+  Map<int, Subscription>? subscriptions;
+  Unreads? unreadsModel;
+
+  @override
+  void onNewStore() {
+    final store = PerAccountStoreWidget.of(context);
+    subscriptions = store.subscriptions;
+
+    unreadsModel?.removeListener(_modelChanged);
+    unreadsModel = store.unreads
+      ..addListener(_modelChanged);
+  }
+
+  @override
+  void dispose() {
+    unreadsModel?.removeListener(_modelChanged);
+    super.dispose();
+  }
+
+  void _modelChanged() {
+    setState(() {
+      // The actual state lives in [subscriptions] and [unreadsModel].
+      // This method was called because one of those just changed.
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    // Design referenced from:
+    //   https://www.figma.com/file/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?type=design&node-id=171-12359&mode=design&t=4d0vykoYQ0KGpFuu-0
+
+    // This is an initial version with "Pinned" and "Unpinned"
+    // sections following behavior in mobile. Recalculating
+    // groups and sorting on every `build` here: it performs well
+    // enough and not worth optimizing as it will be replaced
+    // with a different behavior:
+    // TODO: Implement new grouping behavior and design, see discussion at:
+    //   https://chat.zulip.org/#narrow/stream/101-design/topic/UI.20redesign.3A.20left.20sidebar/near/1540147
+
+    // TODO: Implement collapsible topics
+
+    // TODO(i18n): localize strings on page
+    //   Strings here left unlocalized as they likely will not
+    //   exist in the settled design.
+    final List<Subscription> pinned = [];
+    final List<Subscription> unpinned = [];
+    for (final subscription in subscriptions!.values) {
+      if (subscription.pinToTop) {
+        pinned.add(subscription);
+      } else {
+        unpinned.add(subscription);
+      }
+    }
+    // TODO(i18n): add locale-aware sorting
+    pinned.sortBy((subscription) => subscription.name);
+    unpinned.sortBy((subscription) => subscription.name);
+
+    return Scaffold(
+      appBar: AppBar(title: const Text("Streams")),
+      body: Builder(
+        builder: (BuildContext context) => Center(
+          child: CustomScrollView(
+            slivers: [
+              if (pinned.isEmpty && unpinned.isEmpty)
+                const _NoSubscriptionsItem(),
+              if (pinned.isNotEmpty) ...[
+                _SubscriptionListHeader(context: context, label: "Pinned"),
+                _SubscriptionList(unreadsModel: unreadsModel, subscriptions: pinned),
+              ],
+              if (unpinned.isNotEmpty) ...[
+                _SubscriptionListHeader(context: context, label: "Unpinned"),
+                _SubscriptionList(unreadsModel: unreadsModel, subscriptions: unpinned),
+              ],
+
+              // TODO(#188): add button to "All Streams" page with ability to subscribe
```

Comment:
```suggestion
              // TODO(#188): add button leading to "All Streams" page with ability to subscribe
```

Otherwise this keeps reading to me as about adding to such a page such a button.